web/packages/extension/src/background.ts

```diff
@@ -8,7 +8,138 @@ async function contentScriptRegistered() {
+            ],
+        });
+        needCleanup = true;
+    } catch {
+        return false; // responseHeaders condition not supported.
+    }
+    // Chrome may recognize the properties but have the implementation behind a flag.
+    // When the implementation is disabled, validation is skipped too.
+    try {
+        await utils.declarativeNetRequest.updateDynamicRules({
+            removeRuleIds: [ruleId],
+            addRules: [
+                {
+                    id: ruleId,
+                    condition: { responseHeaders: [] },
+                    action: {
+                        type:
+                            chrome.declarativeNetRequest.RuleActionType
+                                ?.BLOCK ?? "block",
+                    },
+                },
+            ],
+        });
+        needCleanup = true;
+        return false; // Validation skipped = feature disabled.
+    } catch {
+        return true; // Validation worked = feature enabled.
+    } finally {
+        if (needCleanup) {
+            await utils.declarativeNetRequest.updateDynamicRules({
+                removeRuleIds: [ruleId],
+            });
+        }
+    }
+}
+
 async function enable() {
+    // Checks if the responseHeaders condition is supported and not behind a disabled flag.
+    if (utils.declarativeNetRequest && (await isHeaderConditionSupported())) {
+        const playerPage = utils.runtime.getURL("/player.html");
+        const rules = [
+            {
+                id: 1,
+                action: {
+                    type:
+                        chrome.declarativeNetRequest.RuleActionType?.REDIRECT ??
+                        "redirect",
+                    redirect: { regexSubstitution: playerPage + "#\\0" },
+                },
+                condition: {
+                    regexFilter: ".*",
+                    responseHeaders: [
+                        {
+                            header: "content-type",
+                            values: [
+                                "application/x-shockwave-flash",
+                                "application/futuresplash",
+                                "application/x-shockwave-flash2-preview",
+                                "application/vnd.adobe.flash.movie",
+                            ],
+                        },
+                    ],
+                    resourceTypes: [
+                        chrome.declarativeNetRequest.ResourceType?.MAIN_FRAME ??
+                            "main_frame",
+                    ],
+                },
+            },
+            {
+                id: 2,
+                action: {
+                    type:
+                        chrome.declarativeNetRequest.RuleActionType?.REDIRECT ??
+                        "redirect",
+                    redirect: { regexSubstitution: playerPage + "#\\0" },
+                },
+                condition: {
+                    regexFilter: "^.*\\.s(?:wf|pl)(\\?.*|#.*|)$",
```

Comment:
Technically this would match `http://abc.swf#foo` if the `.swf` existed. I guess it's unlikely to ever appear?

Reply:
If you can suggest a more accurate regex I can change this. This was the regex @Dinnerbone added here before it was devided that without the ability to check headers this was insufficient: https://github.com/ruffle-rs/ruffle/pull/12717/files#diff-b9e86d98769e020e68919846919692da73f818952ca6a5b93d84d449af40e916R55

Reply:
This will also only match http://abc.swf#foo if that URL happens to have a Content-Type header of "application/octet-stream", "application/binary-stream", no Content-Type header (as that would match rule 3), or an empty string as a Content-Type header (after adding that to the `values` array). It's far more likely that URL will have a `text/html` Content-Type header, and that's still only if that TLD existed.

Reply:
Thoughts on replacing this with this: `regexFilter: "^.*:\\/\\/.*\\/.*\\.s(?:wf|pl)(\\?.*|#.*|)$"`

That's [anything]://[anything]/[anything].swf(?[anything] or #[anything] or nothing)